src/gallery/slider.js

```diff
@@ -0,0 +1,230 @@
+import { msg } from '../i18n'
+import { requestPageMediaInfo } from '../api'
+
+// internal state of the slider component
+let current = 0,
+	dir = '',
+	lang,
+	gallery,
+	parentContainer
+
+const clientWidth = window.innerWidth,
+	prefixClassname = 'wp-gallery-fullscreen-slider',
+	renderImageSlider = ( images = [], selectedImage = '', givenLang, givenDir, container ) => {
+        const selectedIndex = images.findIndex( image => image.thumb === selectedImage ), // eslint-disable-line
+			imageListHtml = images.map( ( image ) => `
+                <div class="${prefixClassname}-item" style="background-image:url('${image.src}')">
```

Comment:
What's the reasoning behind setting `image.src` as `background-image` here as well as the `src` for the `img` tag?

Reply:
You spot the magic here 💡 

there are two things

1. `image.src` this is hidden by default, and it used for the image load / error event
1. `background-image` this is what user see on the screen with the `background-size: contain` to show the best fit 100% of the image either width or height, but it has a down-side, checkout here for more information and alternative solution https://phabricator.wikimedia.org/T261745

Reply:
Interesting

Reply:
hmmmm, I see there are problems here in **lazy loading image**. Currently the page loads ALL the image in gallery because of the css background image url, that means `<img src="" loading="lazy"/>` doesn't do what we want.

What I can do, manually insert the background image instead of initial all in the render method.